dv/uvm/icache/doc/ibex_icache_dv_plan.md

```diff
@@ -1,99 +1,142 @@
-It provides commonly used handles, variables, functions and tasks that the test sequences can simple use / call.
-Some of the most commonly used tasks / functions are as follows:
-* task 1:
-* task 2:
+
+All test sequences reside in [`dv/uvm/icache/dv/env/seq_lib`](https://github.com/lowRISC/ibex/tree/master/dv/uvm/icache/dv/env/seq_lib).
+
+TBD
 
 #### Functional coverage
-To ensure high quality constrained random stimulus, it is necessary to develop a functional coverage model.
-The following covergroups have been developed to prove that the test intent has been adequately met:
-* cg1:
-* cg2:
+
+TBD
 
 ### Self-checking strategy
 #### Scoreboard
-The `ibex_icache_scoreboard` is primarily used for end to end checking.
-It creates the following analysis ports to retrieve the data monitored by corresponding interface agents:
-* analysis port1:
-* analysis port2:
-<!-- explain inputs monitored, flow of data and outputs checked -->
+The `ibex_icache_scoreboard` is used to check that instructions fetched from the cache lie at the correct addresses and contain the correct data.
+It has two analysis ports: one for the memory agent and another for the core agent.
+
+The scoreboard can keep track of what addresses it expects from responses to the core by modelling the cache's internal address counter and following branches.
+
+To check correctness of the fetched data, the scoreboard will keep track of the set of seeds that have been set by the memory agent since the last cache invalidation began.
+When it sees an instruction being passed back to the core, it can check that the response matches one of the possible seeds.
+
+If the cache is disabled, the scoreboard only checks against seeds valid since the last branch.
+It has to check more than just the most recent seed because the cache's fill buffers aren't invalidated until a branch happens.
+
+If the cache is enabled, the scoreboard must check seeds valid since the last branch that happened at or before the most recent invalidation.
+
+We may wish to make more assertions about the cache's behaviour on the instruction bus.
+For example, that the cache stops making requests shortly after `req_i` goes low.
+To do so, we would just need to monitor fetches in the memory agent as well as seed updates.
 
 #### Assertions
-<!-- only keep this section if any FPV is done on icache -->
+* Interface assertions: TBD
+* Unknown checks on DUT outputs: TBD
 
 ## Building and running tests
-We are using our in-house developed [regression tool]({{< relref "vendor/lowrisc_ip/dvsim" >}}) for building and running our tests and regressions.
-Please take a look at the link for detailed information on the usage, capabilities, features and known issues.
-Here's how to run a basic sanity test:
+
+Tests are built and run with the [`dvsim`](https://github.com/lowRISC/ibex/tree/master/vendor/lowrisc_ip/dvsim) tool (vendored in from the OpenTitan project).
+This is wrapped with a Makefile.
```

Comment:
Is it possible to make this Makefile target agnostic? I'm integrating Ibex with dvsim as well, so it would be good for this Makefile to work with both Ibex/Icache instead of just with one of the two possible targets.
Maybe something like:
```console
$ cd dv/uvm
$ make IP=icache TEST=ibex_icache_sanity
```

Just an idea though, so feel free to propose anything you think might work!

Reply:
I agree that this makes sense in the future, but for now I'd prefer to get each working separately. Let's try to generalise the Makefiles once each works. The easiest way would be adding `dv/uvm/Makefile` and just calling out to the Makefiles below, but maybe we can share more code than that.

Reply:
Sure, this can be worked out in the future.

Reply:
For now, it's just wrapping up the dvsim command line arguments. Let's come back to this when we've actually got a working testbench and are deciding how to present it to potential future users.

Reply:
I'd recommend adding shortcuts in you local bashrc instead. The extra layer doesn't seem to be needed at all.

Reply:
But then everyone who wants to work with the code needs to maintain their own "how to run dvsim.py for testbench foo" shortcuts in their .bashrc. This is wasted effort and will, of course, bit-rot over time.

Alternatively, we could put shortcut commands for "build and run the tests" and similar in the tree. Much better, because they are in one place and will be maintained if the vendored dvsim.py changes. In what format should those commands be stored? Well, I guess we could have a bunch of shell scripts. Alternatively, a Makefile. Which is what I'm suggesting.

Reply:
I dont think any shortcut is needed at all. I dont see how or why 
```
dvsim.py ibex_icache_sim_cfg.hjson -i icache_sanity 
```

Is harder / more complicated than
```
make -i TEST_NAME=icache_sanity
```
All OT DV plans and methodology and getting started docs will be updated to reflect the dvsim invocation (when we get to it), which IMO is sufficient.

Reply:
@sriyerg Is removing mentions of Make a requirement for merging this PR? If not, let's merge it and you or I can make a later PR to use your preferred command line later. If so, please could you make this explicit? Thanks.

Reply:
@sriyerg Thinking again, I've updated the text as requested.